src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/PrisonerService.kt

```diff
@@ -52,16 +53,19 @@ class PrisonerService(
     return prisonApiClient.getPrisonerHousingLocation(prisonerId)
   }
 
-  private fun isPrisonerInTemporaryLocation(levels: List<PrisonerHousingLevelDto>): Boolean {
-    if (levels.isNotEmpty()) {
-      return TransitionalLocationTypes.contains(levels[0].code)
-    }
-    return false
-  }
-
-  fun getLevelsMapForPrisoner(prisonerHousingLocationsDto: PrisonerHousingLocationsDto): Map<PrisonerHousingLevels, String?> {
-    with(prisonerHousingLocationsDto) {
-      val housingLevel = if (isPrisonerInTemporaryLocation(levels)) lastPermanentLevels else levels
+  fun getLevelsMapForPrisoner(prisonerHousingLocation: PrisonerHousingLocationsDto, sessionTemplates: List<SessionTemplate>?): Map<PrisonerHousingLevels, String?> {
+    with(prisonerHousingLocation) {
+      val housingLevel = if (isPrisonerInTemporaryLocation(levels)) {
+        val transitionalLocation = if (levels.isNotEmpty()) levels[0].code else ""
+        // if there are sessions for the prisoners temporary location - level one code needs to be that transitional location
+        if (sessionTemplates != null && hasPrisonGotSessionsWithPrisonersTransitionalLocation(sessionTemplates, transitionalLocation)) {
```

Comment:
if prisoner is in temporary location and there are session templates for that location in the prison show only those sessions.